### PR TITLE
feat(Public ECR): Updated Test Image Name and Added Latest Tag Check

### PR DESCRIPTION
**Pull request by CoshUS**
#### Checklist

- [ ] Add input/output [type hints](https://docs.python.org/3/library/typing.html) to new functions/methods
- [ ] Write design document ([Do I need to write a design document?](https://github.com/aws/aws-sam-cli/blob/develop/DEVELOPMENT_GUIDE.md#design-document))
- [ ] Write unit tests
- [ ] Write/update functional tests
- [ ] Write/update integration tests
- [ ] `make pr` passes
- [ ] `make update-reproducible-reqs` if dependencies were changed
- [ ] Write documentation

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0).

### CODE REVIEW

File: tests/integration/buildcmd/build_integ_base.py [hunk 1]
```diff
@@ -120,10 +120,23 @@ def verify_docker_container_cleanedup(self, runtime):
             time.sleep(1)
         docker_client = docker.from_env()
         samcli_containers = docker_client.containers.list(
-            all=True, filters={"ancestor": "lambci/lambda:build-{}".format(runtime)}
+            all=True, filters={"ancestor": f"public.ecr.aws/sam/build-{runtime}"}
```
Comment (jfuss) [inline tests/integration/buildcmd/build_integ_base.py:123]: ❤️

**Review by jfuss — APPROVED** [review]

**Comment by hoffa** [comment]
💃

File: tests/integration/buildcmd/build_integ_base.py [hunk 2]
```diff
@@ -120,10 +120,23 @@ def verify_docker_container_cleanedup(self, runtime):
...
         )
         self.assertFalse(bool(samcli_containers), "Build containers have not been removed")
 
+    def verify_pulling_only_latest_tag(self, runtime):
```
Comment (hoffa) [inline tests/integration/buildcmd/build_integ_base.py:127]: nit: this doesn't actually check that `latest` was pulled, just that _an_ image was pulled

Comment (CoshUS) [reply tests/integration/buildcmd/build_integ_base.py:127]: Thanks for pointing it out! Added a more explicit check and will be included in a future PR.

**Pull request merged**